services/api/db/serializers.py

```diff
@@ -0,0 +1,101 @@
+from rest_framework import serializers
+from db.models import User
+from django.utils.encoding import smart_str, force_bytes, DjangoUnicodeDecodeError
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.core.validators import validate_email
+
+#Method for validating password.
+def validate_confirm_password(password,confirm_password):
+    if password != confirm_password:
+        raise serializers.ValidationError("Password and confirm password don't match.")
+    return True
+
+
+
+class SignUpEndPointSerializer(serializers.ModelSerializer):
+    confirm_password = serializers.CharField(style={'input_type': 'password'}, write_only=True)
+    class Meta:
+        model = User
+        fields= ['email', 'name', 'password', 'confirm_password']
+        extra_kwargs={
+            'password': {'write_only': True}
+        }
+    #To validate whether the entered email and password are correct.
+    def validate(self, data):
+        email = data.get('email')
+        password, confirm_password = data.get('password'), data.get('confirm_password')
+        try:
+            validate_email(email)
+        except:
+            raise serializers.ValidationError("Please provide a valid email address.")
+        if validate_confirm_password(password,confirm_password):
+            return data
+    def create(self, validated_data):
+        return User.objects.create_user(**validated_data)
+    
+class SignInEndPointSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(
+        max_length=255,
+        style={'input_type': 'email', 'placeholder': 'Email Address'}
+    )
+    class Meta:
+        model = User
+        fields = ['email', 'password']
```

Comment:
is any style to be provided for password field as well ?
for example the style for password has been added in the following serializer:-
```
class UserPasswordResetSerializer(serializers.Serializer):
    password = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)

Reply:
provided styling for password as well

Reply:
but is style required in serializer?

Reply:
its for protecting password view (*** ) , for security reasons. should we be dependent on front end to do so ??